### ISSUE TITLE
Problem with gmail and http request nodes in version 0.139.1.

### ISSUE THREAD

**Issue by Yaccobv2** [n8n-io/n8n#2256]
Hi! There is a problem with gmail node and http request node in version 0.139.1. After I restored n8n to version 0.138.0 everything works perfectly fine.

Gmail node does not apply selected tags and queries correctly, for example node reads emials without "UNREAD" tags or emails that doesn't contain query text "test emails". Also the last item returned by this node is undefined and it repeats itself every time I run the workflow.
Configuration:
```
 {
    "parameters": {
      "resource": "message",
      "operation": "getAll",
      "returnAll": true,
      "additionalFields": {
        "format": "resolved",
        "labelIds": [
          "Label_6577020579567117534",
          "UNREAD"
        ],
        "q": "={{$json[\"0\"][\"Subject\"]}}"
      }
    },
    "name": "Get Messages",
    "type": "n8n-nodes-base.gmail",
    "typeVersion": 1,
    "position": [
      590,
      -40
    ],
    "credentials": {
      "gmailOAuth2": "integromat"
    }
  },
```

Http request doesn't send binary files and results with error:
"Converting circular structure to JSON --> starting at object with constructor TLSSocket | property _httpMessage -> object with constructor ClientRequest --- property socket closes the circle"

Configuration: 
```  
  {
    "parameters": {
      "requestMethod": "POST",
      "url": "=https://api.trello.com/1/cards/{{$node[\"Trello\"].json[\"id\"]}}/attachments?key=***&token=***",
      "allowUnauthorizedCerts": true,
      "jsonParameters": true,
      "options": {
        "bodyContentType": "multipart-form-data",
        "fullResponse": true
      },
      "sendBinaryData": true,
      "binaryPropertyName": "=data"
    },
    "name": "TRELLO ADD ATTACHMENT",
    "type": "n8n-nodes-base.httpRequest",
    "typeVersion": 1,
    "position": [
      2180,
      270
    ],
    "continueOnFail": true
  },
```

Hope you are able to solve this issue or guide me how should I resolve it in my workflow  :) (CC @dolohow)

**Comment by janober** [comment]
Please try again with the just released n8n@0.141.0

It contains multiple fixes

**Comment by todaysday** [comment]
On n8n@0.141.0 and n8n@0.141.1 same problem with Gmail node.
It ignores any Label IDs matching and shows everything in the mailbox

**Comment by janober** [comment]
Thanks for confirming. Did assing to @krynble he will have a look.

**Comment by janober** [comment]
Fix got released with n8n@0.142.0
